model/resolver/resolver.go

```diff
@@ -345,6 +345,43 @@ func (r *Resolver) ResolveLinks() validation.ErrorList {
 		}
 	}
 
+	// Record in each job what roles consume it
+	for _, consumerInstanceGroup := range m.InstanceGroups {
+		for _, consumerJob := range consumerInstanceGroup.JobReferences {
+			for linkName, consumer := range consumerJob.ResolvedConsumers {
+				providerInstanceGroup := m.LookupInstanceGroup(consumer.RoleName)
+				if providerInstanceGroup == nil {
+					// This should not happen: we resolved a link, but can no
+					// longer find the instance group that provides it
+					errors = append(errors, validation.InternalError(
+						fmt.Sprintf(`instance_group[%s].job[%s].consumes[%s]`, consumerInstanceGroup.Name, consumerJob.Name, linkName),
```

Comment:
Declaring variables in a more procedural approach would help reading it. E.g.:
```go
field := fmt.Sprintf("instance_group[%s].job[%s].consumes[%s]", consumerInstanceGroup.Name, consumerJob.Name, linkName)
err := fmt.Errorf("Could not find resolved instance group %s", consumer.RoleName)
internalErr := validation.InternalError(field, err)
errors = append(errors, internalErr)
```

Also, no need for using ` for the strings.